### PR TITLE
feat: add z.looseEnum for flexible enum validation

### LINKED ISSUE #4362: z.looseEnum() for more permissive enum validation
Would it be possible to introduce a `z.looseEnum()` that functions like `z.enum()`, but with more forgiving input validation? In my current workflow, I follow a request/model/response schema pattern. While models and responses are strictly typed, I want my request objects to be more permissive. Currently, I work around this by doing something like:

```ts
z.enum(['foo', 'bar']).or(z.string())
```

A dedicated `z.looseEnum()` could allow unknown values through in request schemas.

**Examples:**

**Request:**

```ts
const RequestSchema = z.looseObject({
  status: z.looseEnum(['pending', 'cancelled', 'done']),
});
```

**Model:**

```ts
const ModelSchema = z.strictObject({
  status: z.enum(['pending', 'cancelled', 'done']),
});
```

**Response:**

```ts
const ResponseSchema = z.strictObject({
  status: z.enum(['pending', 'cancelled', 'done']),
});
```

### PR DESCRIPTION
**Pull request by Lavdewangan1**
# Add `z.looseEnum()` for permissive enum validation

## Fixes #4362  - Request for permissive enum validation

Introduces a new `z.looseEnum()` validator that provides more flexible input validation while maintaining type safety.

### Key Features
- ✅ Preserves TypeScript type hints for preferred values
- ✅ Accepts any string input (unlike strict `z.enum()`)
- 🔗 Seamless integration with other Zod validators
- 🏗 Designed for permissive request validation

## Usage

```typescript
// Basic usage
const StatusSchema = z.looseEnum(['pending', 'completed']);
StatusSchema.parse('pending');  // valid
StatusSchema.parse('other');    // also valid
StatusSchema.parse(123);        // throws ZodError

// With object schemas
const RequestSchema = z.object({
  status: z.looseEnum(['on', 'off']),
  id: z.string()
});

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Introduced a new schema type, `looseEnum`, which validates any string input, allowing both predefined enum values and arbitrary strings.
- **Bug Fixes**
  - None.
- **Documentation**
  - Updated sponsor link in the README file.
- **Tests**
  - Added comprehensive tests for the new `looseEnum` schema, including integration and edge cases.
  - Suppressed TypeScript warnings for unused imports in test files.
- **Chores**
  - Upgraded development dependencies and added new ones in the project configuration.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

### CODE REVIEW

**Comment by RobinTail** [comment]
@Lavdewangan1 ,

why `z.enum().or(z.string())` does not suite the need?

File: package.json [hunk 1]
```diff
@@ -118,5 +119,8 @@
       "npm-funding": true
     }
   },
-  "types": "./index.d.ts"
+  "types": "./index.d.ts",
+  "dependencies": {
+    "yarn": "^1.22.22"
+  }
```
Comment (RobinTail) [inline package.json:125]: this does not look right

```suggestion
  "types": "./index.d.ts"
```

**Comment by Lavdewangan1** [comment]
@RobinTail 
z.enum([...]).or(z.string()) works as a workaround, but it's not ideal. It loosens the type completely and makes the intent less clear. A dedicated z.looseEnum() would be more expressive, keep things cleaner, and better support patterns where requests are permissive but models/responses are strict.

**Comment by samchungy** [comment]
I would think that you ideally want all the basic string methods to also apply to this right?

Essentially a string validator + some extra enum options you specify to narrow the type down?

**Comment by Lavdewangan1** [comment]
@samchungy ohh yaah, it should
